node/secret.py

```diff
@@ -0,0 +1,165 @@
+            self.public_key = serialization.load_pem_public_key(
+                public_pem,
+                backend=default_backend()
+            )
+
+        logger.info("RSA keys loaded from files.")
+
+    def get_public_key(self):
+        public_numbers = self.public_key.public_numbers()
+
+        # Using a fixed `kid` that stays the same for this key
+        jwk = {
+            "kty": "RSA",
+            "kid": "server-key",
+            "use": "enc",
+            "n": self._encode_base64_url(public_numbers.n),
+            "e": self._encode_base64_url(public_numbers.e),
+        }
+
+        return jwk
+
+    def check_and_generate_aes_secret(self):
+        if not self._aes_secret_exists():
+            logger.info("AES secret not found. Generating new AES secret...")
+            self.generate_aes_secret()
+        else:
+            logger.info("AES secret already exists in .env.")
+
+    def _aes_secret_exists(self):
+        if not os.path.exists(self.env_file):
+            return False
+        
+        with open(self.env_file, "r") as f:
+            for line in f:
+                if line.startswith("AES_SECRET="):
+                    return True
+        return False
+
+    def generate_aes_secret(self):
+        aes_key = os.urandom(32)
+        
+        self.save_to_env("AES_SECRET", base64.b64encode(aes_key).decode())
+        logger.info("AES secret generated and saved to .env.")
+
+    def decrypt_rsa(self, base64_message):
+        encrypted_message = base64.b64decode(base64_message)
+
+        decrypted_message = self.private_key.decrypt(
+            encrypted_message,
+            asymmetric_padding.OAEP(
+                mgf=asymmetric_padding.MGF1(algorithm=hashes.SHA256()),
+                algorithm=hashes.SHA256(),
+                label=None
+            )
+        )
+
+        return decrypted_message.decode('utf-8')
+
+    def encrypt_with_aes(self, data: str, aes_key: bytes):
+        iv = os.urandom(16)
+        padder = padding.PKCS7(128).padder()  # 128 bits = 16 bytes
+        padded_data = padder.update(data.encode()) + padder.finalize()
+
+        cipher = Cipher(algorithms.AES(aes_key), modes.CBC(iv), backend=default_backend())
+        encryptor = cipher.encryptor()
+        encrypted_data = encryptor.update(padded_data) + encryptor.finalize()
+
+        return base64.b64encode(iv + encrypted_data).decode()
+
+    def decrypt_with_aes(self, encrypted_data: str, aes_key: bytes):
+        encrypted_data = base64.b64decode(encrypted_data)
+        iv = encrypted_data[:16]
+        encrypted_message = encrypted_data[16:]
+        cipher = Cipher(algorithms.AES(aes_key), modes.CBC(iv), backend=default_backend())
+
+        decryptor = cipher.decryptor()
+        decrypted_data = decryptor.update(encrypted_message) + decryptor.finalize()
+        
+        unpadder = padding.PKCS7(128).unpadder()
+        original_data = unpadder.update(decrypted_data) + unpadder.finalize()
```

Comment:
have you considered using `GCM`? It provides message authentication so you can know if it the secret was tampered with, which aside from the security benefits is nice because it can help you avoid implementation errors